src/gui/src/tclCmdHighlighter.h

```diff
@@ -32,40 +32,57 @@
 
 #pragma once
 
-#include <tcl.h>
-
+#include <memory>
 #include <set>
+#include <string>
+#include <tuple>
 #include <vector>
-#include <memory>
 
-#include <QSyntaxHighlighter>
 #include <QRegularExpression>
+#include <QSyntaxHighlighter>
+#include <QTextBlockUserData>
 #include <QTextCharFormat>
 
 namespace gui {
 
 using QRegularExpressionPtr = std::unique_ptr<QRegularExpression>;
-struct SyntaxRule
+struct CommandRule
 {
   QRegularExpressionPtr pattern;
-  std::unique_ptr<std::vector<QRegularExpressionPtr>> args;
+  int command; // command index
```

Comment:
index into what?

Reply:
I added some additional wording. it's the index for the commands_ vector in tclCmdInputWidget.